### PR TITLE
Add DAO entityID for sendSMSMessage() $smsProviderParams

### PR DESCRIPTION
**Pull request by munkeywolf00**
Overview
----------------------------------------
When making a CaseActivity for a scheduled reminder, it would help if we can include entityID in $smsProviderParams to be able to trace which CaseActivity was related to a SMS delivery where `result = $apiMsgID` resides. With the added $smsProviderParams, we can use it that could help track whether an Inbound SMS is related to a CaseActivity from a scheduled reminder.

Before
----------------------------------------
Unable to track which CaseActivity a SMS delivery is related to.

After
----------------------------------------
Help track which CaseActivity a SMS delivery is related to using the DAO entityID

Comments
----------------------------------------
CiviCRM: 5.71.1
Here is a scenario:
Make CaseActivity, Scheduled Reminder Runs, `Outbound SMS` is created, `SMS delivery` is created, `Reminder Sent` is created.
We are looking into connecting `Inbound SMS` to the CaseActivity. Since we have `$apiMsgId` `result` in `Inbound SMS`, we can find out which `SMS delivery` it is connected, however, we cannot find out which CaseActivity it is connected. 

EntityID may not be the CaseActivity, but we devs can just query to check it.

### CODE REVIEW

File: CRM/Activity/BAO/Activity.php [hunk 1]
```diff
@@ -1319,6 +1321,7 @@ public static function sendSMSMessage(
     $recipient = $toPhoneNumber;
     $smsProviderParams['contact_id'] = $toID;
     $smsProviderParams['parent_activity_id'] = $activityID;
+    $smsProviderParams['dao_entity_id'] = $entityID;
```
Comment (eileenmcnaughton) [inline CRM/Activity/BAO/Activity.php:1324]: I think drop the `dao` and change to just `entity_id` - seems more like other patterns

Comment (munkeywolf00) [reply CRM/Activity/BAO/Activity.php]: Thank you for the feedback. I removed it now

**Comment by eileenmcnaughton** [comment]
This seems OK to me - can you do another PR to add yourself to our contributors file https://github.com/civicrm/civicrm-core/blob/master/contributor-key.yml

**Comment by devdavc** [comment]
I already my name in there. So it is fine to not make one.

**Comment by eileenmcnaughton** [comment]
erm @devdavc - are you @munkeywolf00 also?

**Comment by devdavc** [comment]
Yes. Sorry.

**Comment by eileenmcnaughton** [comment]
Ok that was confusing

**Pull request merged**